lib/inc/internal/ruby/api.hpp

```diff
@@ -21,6 +21,8 @@ namespace facter { namespace facts {
 
 namespace facter {  namespace ruby {
 
+namespace lth_dl = leatherman::dynamic_library;
```

Comment:
We shouldn't alias namespaces in headers if we can avoid it.  I agree that `leatherman::dynamic_library::dynamic_library` is a bit verbose and redundant, but we could alias it properly in the source file and only have to use the fully-qualified name a few times in the header.  If we do prefer to alias a namespace, I'd prefer something like `lm_dl`.  `lth` looks a little strange to me.